libmamba/include/mamba/core/pool.hpp

```diff
@@ -51,9 +38,24 @@ namespace mamba
         void remove_repo(Id repo_id);
 
     private:
-        std::pair<spdlog::logger*, std::string> m_debug_logger;
-        Pool* m_pool;
-        std::list<MRepo> m_repo_list;
+        struct MPoolData;
+
+        Pool* pool();
+        Pool const* pool() const;
+
+        /**
+         * Make MPool behave like a shared_ptr (with move and copy).
+         *
+         * The pool is passed to the ``MSolver`` for its lifetime but the pool can legitimely
+         * be reused with different solvers (in fact it is in ``conda-forge``'s ``regro-bot``
+         * and ``boa``).
+         * An alternative considered was to make ``MPool`` a move-only type, moving it in and out
+         * of ``MSolver`` (effectively borrowing).
+         * It was decided to make ``MPool`` share ressources for the following reasons.
+         *    - Rvalue semantics would be unexpected in Python (and breaking ``conda-forge``);
+         *    - Facilitate (potential) future investigation of parallel solves.
+         */
+        std::shared_ptr<MPoolData> m_data;
```

Comment:
Didnt  we conclude that `MPool` would first have a move-only behavior (and implemented with `unique_ptr`), then later would be switched to shared when we will need it? You went directly with shared because you found cases where it was necessary already?

Reply:
No, for the reasons mentioned here we later decided to go that way.